File: flight-search-system-backend/src/main/java/com/flightsearch/entity/FlightFilterDto.java
package com.flightsearch.entity;

import java.time.LocalDateTime;
import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class FlightFilterDto {

    private Integer originAirportId;
    private Integer destinationAirportId;
    private LocalDateTime deptDate;
    private LocalDateTime returnDate;
    private FlightType flightType;
    private TravelClass travelClass;
    private List<Integer> aircrafts;
    private List<Integer> airlines;
    private Integer layovers;
    private Double maxPrice;
    private Integer time;
}

File: flight-search-system-frontend/src/pages/api/types/flightFilterDto.ts
import { FlightType } from "./flightType";
import { TravelClass } from "./travelclass";

export interface FlightFilterDto {
    originAirportId?: number | null;
    destinationAirportId?: number | null;
    deptDate?: string | null;
    returnDate?: string | null;
    flightType?: FlightType |string |null;
    travelClass?: TravelClass |null |string;
    aircrafts?: number[];
    airlines?: number[] |null;
    layovers?: number;
    maxPrice?: number;
    time?: number;
}